src/Illuminate/Database/Eloquent/Relations/Relation.php

```diff
@@ -13,6 +13,9 @@
 use Illuminate\Support\Traits\ForwardsCalls;
 use Illuminate\Support\Traits\Macroable;
 
+/**
+ * @template T of \Illuminate\Database\Eloquent\Model
```

Comment:
This generic name (`T`) isn't specific enough and should be indicative of its purpose. It's also a bad idea for child classes to override the template name and this will cause issues downstream as sometimes we grab the generic by name---introducing a name change in the inheritance chain adds unnecessary complexity.

The Larastan stubs use: 
```suggestion
 * @template TRelatedModel of \Illuminate\Database\Eloquent\Model
```